contrib/parseq-retry/src/main/java/com/linkedin/parseq/retry/RetryFailureException.java

```diff
@@ -0,0 +1,17 @@
+package com.linkedin.parseq.retry;
+
+/**
+ * Exception that is thrown when the process of retries is interrupted by any reason.
+ * Usually it's either the exhausted number of attempts, or the unacceptable task result.
+ */
+public class RetryFailureException extends Exception {
+  private static final long serialVersionUID = 0L;
+
+  public RetryFailureException() {
+    super();
+  }
+
+  public RetryFailureException(final String message) {
+    super(message);
+  }
```

Comment:
But outcome.getError() is of `Throwable` type so it will already include stack trace of the actual task error. It's a bad idea to wrap it in extra `RetryFailureException` because that would essentially hide common task failures like `RestliException` and would always require unpacking `RetryFailureException` by callers who wish to use retry wrapper.

Existing `new RetryFailureException` on line 160 is on the right side of ternary operator, so it means there is no `Throwable` available from the task and there's nothing to pass to that new constructor.